packages/roosterjs-editor-dom/lib/table/tableFormatInfo.ts

```diff
@@ -1,6 +1,59 @@
+import { getMetadata, setMetadata } from '../metadata/metadata';
 import { TableFormat } from 'roosterjs-editor-types';
+import {
+    createBooleanDefinition,
+    createNumberDefinition,
+    createObjectDefinition,
+    createStringDefinition,
+} from '../metadata/definitionCreators';
 
-const TABLE_STYLE_INFO = 'roosterTableInfo';
+const TableFormatMetadata = createObjectDefinition<Required<TableFormat>>(
+    {
+        topBorderColor: createStringDefinition(
+            false /** isOptional */,
+            undefined /** value */,
+            true /** allowNull */
+        ),
+        bottomBorderColor: createStringDefinition(
+            false /** isOptional */,
+            undefined /** value */,
+            true /** allowNull */
+        ),
+        verticalBorderColor: createStringDefinition(
+            false /** isOptional */,
+            undefined /** value */,
+            true /** allowNull */
```

Comment:
All these 3 have the same value, you can create a common one first:

const NullableStringDef = createStringDefinition(false, undefined, true);

const TableFormatMetadata = { topBordercolor: NullableStringDef , bottomBorderColor: NullableStringDef , ...